### PR TITLE
Jonas-Week06 (1778099, 1623571)

### PR DESCRIPTION
**Pull request by DrKeks1010**
Taks 1:
the total sum of all directories smaller than 100000 is 1778099

Task 2:
To make space for the file with a size of 30000000, something greater than 1609574 has to be deleted:
Found directory: bwmglvmt at \pcqjnl\lrrl\nwjggvr\bwmglvmt with a size of 1623571

### CODE REVIEW

File: Exercises/JonasPutz/exercises_06.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    delete_size: int = required_space + dir_tree.get_size() - available_space
+
+    if delete_size <= 0:
+        print("No directory needs to be deleted!")
+        return
+    
+    print(f"Required space to be delete: {delete_size}") 
+    print(f"Found directory: {dir_tree.find_to_delete(delete_size)[0]}")
+    
+delete_dir(_directory_tree, 30000000, 70000000)
```
Comment (kuckuk3) [inline Exercises/JonasPutz/exercises_06.py:226]: no magic values

**Review by kuckuk3 — CHANGES_REQUESTED** [review]

File: Exercises/JonasPutz/exercises_06.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        return self._parent
+    
+    def get_sub_directory(self, name: str):
+        # returnes a subdirectory with the specified name
+        for dir in self._dirs:
+            if dir.name == name:
+                return dir
+            
+        # didnt find subdirectory, an error will be raised
+        raise NameError(f"Unknown directory {name}")
```
Comment (ClasherKasten) [inline Exercises/JonasPutz/exercises_06.py:133]: A more pythonic way would be to use `@property`.

File: Exercises/JonasPutz/exercises_06.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    
+_directory_tree : Directory = Directory('/', None, Path())
+
+def build_dir_tree(file_path):
+    current_dir = _directory_tree
+    with file_path.open("r") as ifile:
+        for line in (line.strip() for line in ifile):
+            if line.startswith('$ '): 
+                # line is a command
+                line = line[2::]
```
Comment (ClasherKasten) [inline Exercises/JonasPutz/exercises_06.py:172]: Only `line[2:]` is needed.

**Review by ClasherKasten — CHANGES_REQUESTED** [review]
Pretty nice solution, especially the type annotations are a big plus :+1:

**Review by RubelOS — APPROVED** [review]
Really good solution! Well done!

**Pull request merged**